File: python/Venniyammal/Inheritance and classes/lorry.py
from Vehicle import *

#class to define lorry methods
class lorry(Vehicle):
    def lorryexecution():
        ch=0
        speed = 10
        maxspeed=150
        f=False
        while(ch !=4):
            try:
                
                
                print("Provide me an instruction to drive ")
                ch=int(input("\n\t\t 1.To start 2.To increase the  speed 3.To turn 4.To stop the lorry "))
                if(ch==1 and f==False):
                    v=Vehicle("lorry")
                    print("close the doors !!! wear your seat belts")
                    print("lorry has starte at the speed of ",speed)
                    f=True
                elif(ch == 2 ):
                    speed=speed+20
                    if(speed<=maxspeed):
                        print("Now the lorry is running at the speed of ",speed)
 
                    else:
                        print("Running at the Maximum speed" )
                        speed=speed-20
                        continue
                elif(ch==3 and speed<=maxspeed):
                    speed=speed-30
                    if(speed<10):
                        print ("Running at initial speed")
                    else:
                        v.turn("lorry")
                        print("Now the Speed is ",speed )
                        speed=speed+10
                        print("turn completed now the bike is running at the speed",speed)
                elif(ch==4):
                    v=Vehicle("Lorry")
                    v.stop("lorry")
            except ValueError:
                print("Enter the valid choice")
                continue

        
            

File: Venniyammal/Workspace/Assesment2/src/com/treselle/Lorry.java
package opt.com.treselle;
//venni

import java.util.Scanner;
class Lorry extends sound implements vehicle , fourwheeler
    {
     static int speed= 10;
	 int maxSpeed = 50;
	 int limit =40;
	 
	 // creating function for Increase and Decrease the Speed
	 
	 void increaseSpeed() 
	    { 
		    Exception e = new Exception();
	         speed = 10+speed;
		    /* while(limit!=speed)
		    {
		     System.out.println("\n \t Now the bike is running at the Speed of " + speed + " Km/h.");
			}*/
		System.out.println("\n \t Now the bike is running at the Speed of " + speed + " Km/h.");
		  if(speed==maxSpeed)
		    {
			 System.out.println(" Alert !!! Alert!!! Your Exceeding the speed limit");
			}
		   else if ( speed>=limit)
			{
			 System.out.println("\n\t Your running at the 80% of the speed");
			 }
		    
	    }
	 void decreaseSpeed()
	    {
	         speed =speed-20;
		     System.out.println("\n\t Now the Lorry is running at the Speed of " + speed +"  Km/h." );
		}
		
		
	 // implementation of Interface methods 
	  void horn()
        {
         System.out.println("\n \t Beeeeeeeep");
        }
	 
	 public void openDoor()
	    {
		     System.out.println("\n \t Open the Door");
		}
	 public void closeDoor()
	    {
		     System.out.println("\n\t Close the Door");
		}
		
	 public void start()
	     {
			 openDoor();
			 closeDoor();
		     System.out.println("\n \t Now the Lorry Started at the Initial Speed of 10 km/h");
		 }
	 public void turn()
	     { 
		    
		     decreaseSpeed();
		     horn();
		     System.out.println( " \n \t Turn Completed , and Increasing the Speed");
		     increaseSpeed();
		  }
	 public void stop()
		 {
		    while(speed!=10)
			     { 
				     decreaseSpeed();
				 }
		    System.out.println("\n \t Now the Lorry is stopped ");
			openDoor();
			closeDoor();
		 }
		 
		Scanner in = new Scanner(System.in);
		public void excecute()
		{
		int n =1;
     	while(n!=5)
		{
		   
			System.out.println("\n  *!*!*!*! provide me an instuction to drive the car.");
			System.out.println("\n \n \t \t Press 1 : To Start \n\t\tPress 2 : To increase the speed \n\t\t Press 3 : To return \n\t\tPress 4 : To Stop");
			
            int ch = in.nextInt();
			switch(ch)
			{
			case 1:
				start();
				break;
			case 2:
			    increaseSpeed();
				break;
			case 3:
				turn();
				break;
			case 4:
				stop();
				break;
			}
		n=ch;
		
		}
     }
	}
	